src/ray/core_worker/core_worker.cc

```diff
@@ -704,9 +704,9 @@ bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle) {
           it->second->Reset();
         }
       } else if (actor_data.state() == gcs::ActorTableData::DEAD) {
-        RAY_CHECK_OK(gcs_client_->Actors().AsyncUnsubscribe(actor_id, nullptr));
         // We cannot erase the actor handle here because clients can still
-        // submit tasks to dead actors.
+        // submit tasks to dead actors. This also means we defer unsubscription,
```

Comment:
I also found this issue debugging a flaky test.

Reply:
I'm a bit confused, what is the crash that occurs during the unsubscribe?

Reply:
The crash is when we iterate over actor handles to delete them all in
SetCurrentTaskId().

Reply:
Sorry I still don't understand completely, why does that crash happen?

Reply:
You can't AsyncUnsubscribe() twice. It will crash the second time with "subscription not found".